### PR TITLE
Feat/update tldts

### PR DESCRIPTION
**Pull request by amebosc**
- **Update tldts:** the latest tldts allows to parse urls without extension. Let's remove the use of the URL api introduced in the latest commit
- **Resolve npm audit issues:** update mocha and ts-node to latest versions

### CODE REVIEW

File: ts/lib/url-parser.ts [hunk 1]
```diff
@@ -123,20 +117,15 @@ export function extractRootDomain(url: string): string {
         return null;
     }
     const parsedUrl = parse(url);
-    if (domainIsIP(parsedUrl.hostname) || parsedUrl.hostname === 'localhost') {
+    if (domainIsIP(parsedUrl.hostname) // the url is an IP address
+        || parsedUrl.hostname === 'localhost' // the url is a localhost
+        || parsedUrl.hostname === parsedUrl.publicSuffix) { // the url has no extension (ex. http://url-without-extension)
```
Comment (PLhery) [inline ts/lib/url-parser.ts:122]: I think you can replace the whole function with

return domain || hostname || null;

(This condition is always true if there's no domain ; if not we have a weird edge case)

**Review by PLhery — CHANGES_REQUESTED** [review]

File: ts/lib/url-parser.ts (hunk 1 above)
Comment (dhenot) [reply ts/lib/url-parser.ts:122]: > I think you can replace the whole function with
>
> return domain || hostname || null;

I'm not sure that's true ... are there cases where domain is not null, is not the same as hostname, and hostname===publicSuffix ? That would be a case where the result would be different. But I'm really unsure if that's possible or not.

**Review by dhenot — APPROVED** [review]

File: ts/lib/url-parser.ts (hunk 1 above)
Comment (PLhery) [reply ts/lib/url-parser.ts:122]: "are there cases where domain is not null, is not the same as hostname, and hostname===publicSuffix ? "

"domain is not null" => so domain = xxxxx.yyyy, hostname = (zzzz?.)xxxx.yyyy

"and hostname == publicSuffic" => zzzz.xxxx.yyyy == yyyy

I don't think it's possible

You can easily play with tldts on https://npm.runkit.com/tldts to try any weird cases.

Comment (PLhery) [reply ts/lib/url-parser.ts:122]: For instance, if we parse a public suffix as a hostname, domain === null

image

**Review by PLhery — APPROVED** [review]

File: ts/lib/url-parser.ts (hunk 1 above)
Comment (PLhery) [reply ts/lib/url-parser.ts:122]: If you's like to let's merge, and I'll open a follow-up PR to simplify the whole file tomorrow

**Review by highfivedenis — APPROVED** [review]

File: ts/lib/url-parser.ts (hunk 1 above)
Comment (PLhery) [reply ts/lib/url-parser.ts:122]: Actually I opened it here https://github.com/Dashlane/url-parser/pull/21 (it was ready on friday but I didn't want to open a PR on a friday evening)

Comment (amebosc) [reply ts/lib/url-parser.ts:122]: Thanks for your comments! Yes I need to merge this asap, but for sure let's discuss this tomorrow. I think @PLhery  is right, I can't think of cases when this is not true.

**Pull request merged**